### DISCUSSION TITLE
Morphia and Indexes

### DISCUSSION THREAD

**Discussion by vSKAH** [MorphiaOrg/morphia#2855] [General]
>  Please ask your question and provide whatever background or supporting information you think might be relevant: 

Hello I did this, however when I look in mongo compass the indexes do not appear. Do you have an idea? 

 
```JAVA
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(value = "player_battlepass")
@Indexes({
        @Index(fields = {@Field("player_id"), @Field("battlepass_id")}, options = @IndexOptions(unique = true, background = true, name = "player_battlepass_index")),
        @Index(fields = @Field("battlepass_id"), options = @IndexOptions(unique = true, background = true, name = "battlepass_id_index"))
})
public class PlayerBattlePass {

    @Id
    private int id;

    @Property("player_id")
    private UUID playerId;

    @Property("battlepass_id")
    private int battlePassId;
    
}

```

![image](https://github.com/MorphiaOrg/morphia/assets/68877411/3af86041-32f7-4e09-b0f5-8becae74d9e6)

>    Server Version: MongoDB 5.0.24 Community
    Driver Version: 4.11.0
    Morphia Version: 2.4.11

**Comment by evanchooly** [comment]
did you call `datastore.ensureIndexes()` ?

**Reply by vSKAH** [reply]
No, sorry i don't use because in 2.4.11 the fonction deprecated :/

**Reply by vSKAH** [reply]
Don't you have another way of doing it?

**Reply by evanchooly** [reply]
It's done automatically in most cases with 2.4.11 and the new config system but you can call it directly if you'd like.  That method just doesn't exist in 3.0 anymore.

**Reply by oewang** [reply]
@evanchooly I have the same problem, but I call the `datastore.ensureIndexes()` method and the indexes are not generated.

MongoDB Server Version: 4.0.23
Driver Version: 3.4.2
Morphia Version: 1.3.2

**Reply by evanchooly** [reply]
I would recommend upgrading to 1.6 at a minimum 2.4 if possible.  1.3.2 is rather ancient and has been working for a decade or so.  Double check that you're looking in the correct database, etc., for your indexes. Check the logs for any errors.
